Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -3859,20 +3882,67 @@ private ElementInfo GetElementInfoWithIWebElement(IWebElement el, HtmlNode elNod
             if (setFullElementInfoDetails)
             {
                 EI.RelXpath = mXPathHelper.GetElementRelXPath(EI);
-                EI.ElementName = GetBestElementName(EI);
-                EI.Locators = ((IWindowExplorer)this).GetElementLocators(EI);
-                ((IWindowExplorer)this).UpdateElementInfoFields(EI);
-                EI.Properties = ((IWindowExplorer)this).GetElementProperties(EI);
+                EI.ElementName = GetElementName(EI);
+                EI.Locators = ((IWindowExplorer)this).GetElementLocators(EI); 
+                ((IWindowExplorer)this).UpdateElementInfoFields(EI); 
+                EI.Properties = ((IWindowExplorer)this).GetElementProperties(EI); 
             }
 
             return EI;
         }
 
-        string GetBestElementName(ElementInfo EI)
+        string GetElementName(HTMLElementInfo EI)
```

Comment:
you modified the original "GetBestElementName" function which now based on "HTMLElementObject" just need to make sure other calls won;'t get to this function with HTMLElementObject null

Reply:
corrected,
I added solution to support old functionality as well